chucks-pick3-web/lib/bootstrapFormValLogin.js

```diff
@@ -19,38 +33,101 @@
 
 
+<<<<<<< HEAD
+
+function checkPassWord(input) {
+=======
 function checkedUserName(input) {
+>>>>>>> 889aa67190330482f559c395c9a4a39b72cb8007
     if (input !== null && input !== undefined) {
         setErrorMessage(".invalid-feedback." + input.name, "", input);
 
         if (input.required === true) { // the input is required if not show error message // must have required as a input field
             if (hasData(input)) { // input can not be null or undefined it will throw a error
-                checkUserNameValidity(input);
+                checkPassValidity(input);
             } else {
                 setErrorMessage(".invalid-feedback." + input.name, input.name.toNormalCase() + " is required and can not be empty", input);
             }
         } else {
             if (hasData(input)) { // input can not be null or undefined it will throw a error
-                checkUserNameValidity(input);
+                checkPassValidity(input);
             }
         }
     }
 
 
 
+<<<<<<< HEAD
+    function checkPassValidity(input) {
+        const errorFieldName = input.value.toNormalCase();
+        var PasswordContainsUpperCase = /[A-Z]/g;//test for uppercase letter
+        var PasswordContainsLowerCase = /[a-z]/g; //test for lowercase letter
+        var PasswordContainsNumber = /[0-9]/g; //test for number
+        var PasswordContainsSpecialChar = /\W/g; //test for special character
+=======
 }
 
+>>>>>>> 889aa67190330482f559c395c9a4a39b72cb8007
+
+        if(!input.value.match(PasswordContainsUpperCase)) {
+            setErrorMessage(".invalid-feedback." + input.name, errorFieldName + " Password must contain a Uppercase letter", input);
+            return false;
+        }
+        if(!input.value.match(PasswordContainsNumber)) {
+            setErrorMessage(".invalid-feedback." + input.name, errorFieldName + " Password must contain a number", input);
+            return false;
 
-//*********************LOGIN USERNAME & PASSWORD VALIDATION**************************
+<<<<<<< HEAD
+        }
+        if(!input.value.match(PasswordContainsLowerCase)) {
+            setErrorMessage(".invalid-feedback." + input.name, errorFieldName + " Password must contain a lowercase character", input);
+            return false;
+        }
+        if(!input.value.match(PasswordContainsSpecialChar)) {
+            setErrorMessage(".invalid-feedback." + input.name, errorFieldName + " Password must contain a special character", input);
+            return false;
+        }
+        if (!input.value.length >= 8) {
+            setErrorMessage(".invalid-feedback." + input.name, errorFieldName + " Password must contain 8 characters", input);
+            return false;
+        }
+}
 
+=======
+>>>>>>> 889aa67190330482f559c395c9a4a39b72cb8007
 
+function checkUserName(input) {
 
+<<<<<<< HEAD
+    if (input !== null && input !== undefined) {
```

Comment:
Do same checks like in checkStateButton() method above; but instead of checking input.value you need to check that input.name has a value via the `if (input.name.length > 0) ` if statement.